### DISCUSSION TITLE
Global service account API token

### DISCUSSION THREAD

**Discussion by pschmitt** [kubermatic/kubermatic#10183] [Q&A]
Hello,

I'm working on quick and dirty cli wrapper for the API.
I haven't found a way to create an API token that can access all projects my specific user has access to. Service Account Tokens don't really do it since they are inherently bound to a project.

Is there a way to generate an API token that has access to a multitude of projects?

Thanks!

**Comment by moadqassem** [comment]
Just of curiosity, why not using the kubectl directly instead of building this cli wrapper? By the end of the day, whatever you can do in using the KKP RestAPI you can also do using kubectl if you have the seed kubeconfig.

**Reply by pschmitt** [reply]
I'm just a customer of the KKP instance, I only have access to some clusters, but not all of them, especially not the seed cluster.

**Reply by moadqassem** [reply]
I see, so from an API point of view, we use OIDC to access the dashboard and the api. Once you are authenticated you get a JWT token, this token decides the degree of your authorization. You can use this token to create/view/update whatever resources you have access to.

**Reply by pschmitt** [reply]
That's what I ended up doing in my script.
Feels quite a bit hacky and will certainly break if my provider decides to update the OIDC login process for some reason. Would have been nicer to just get a token directly from the KKP interface instead;)
